src/com/owncloud/android/ui/activity/ReceiveExternalFilesActivity.java

```diff
@@ -476,124 +471,40 @@ private boolean somethingToUpload() {
-                copyThenUpload(contentUris.toArray(new Uri[contentUris.size()]),
-                    contentRemotePaths.toArray(new String[contentRemotePaths.size()]));
+        if (resultCode == UriUploader.UriUploaderResultCode.OK) {
+            finish();
+        } else {
 
-            } else if (schemeFileCounter == 0) {
-                showErrorDialog(
-                    R.string.uploader_error_message_no_file_to_upload,
-                    R.string.uploader_error_title_no_file_to_upload
-                );
+            int messageResTitle = R.string.uploader_error_title_file_cannot_be_uploaded;
+            int messageResId = R.string.common_error_unknown;
 
-            } else {
-                finish();
+            if (resultCode == UriUploader.UriUploaderResultCode.ERROR_NO_FILE_TO_UPLOAD) {
+                messageResId = R.string.uploader_error_message_no_file_to_upload;
+                messageResTitle = R.string.uploader_error_title_no_file_to_upload;
+            } else if (resultCode == UriUploader.UriUploaderResultCode.ERROR_READ_PERMISSION_NOT_GRANTED) {
+                messageResId = R.string.uploader_error_message_read_permission_not_granted;
+            } else if (resultCode == UriUploader.UriUploaderResultCode.ERROR_UNKNOWN) {
+                messageResId = R.string.common_error_unknown;
             }
 
-        } catch (SecurityException e) {
-            Log_OC.e(TAG, "Permissions fail", e);
             showErrorDialog(
-                R.string.uploader_error_message_read_permission_not_granted,
-                R.string.uploader_error_title_file_cannot_be_uploaded
+                    messageResId,
+                    messageResTitle
             );
-
-        } catch (Exception e) {
-            Log_OC.e(TAG, "Unexpted error", e);
-            showErrorDialog(
-                R.string.common_error_unknown,
-                R.string.uploader_error_title_file_cannot_be_uploaded
-            );
-
-        } finally {
-            // Save the path to shared preferences; even if upload is not possible, user chose the folder
-            PreferenceManager.setLastUploadPath(mUploadPath, this);
         }
     }
 
-
-    private String generateDiplayName() {
-        return getString(R.string.common_unknown) +
-            "-" + DisplayUtils.unixTimeToHumanReadable(System.currentTimeMillis());
-    }
-
-    /**
-     *
-     * @param sourceUris        Array of content:// URIs to the files to upload
-     * @param remotePaths       Array of absolute paths to set to the uploaded files
-     */
-    private void copyThenUpload(Uri[] sourceUris, String[] remotePaths) {
-        showWaitingCopyDialog();
-
-        CopyAndUploadContentUrisTask copyTask = new CopyAndUploadContentUrisTask(this, this);
-        FragmentManager fm = getSupportFragmentManager();
-        TaskRetainerFragment taskRetainerFragment =
-            (TaskRetainerFragment) fm.findFragmentByTag(FTAG_TASK_RETAINER_FRAGMENT);
```

Comment:
This code is being lost; this is important, we can have problems in rotations. Need to have it also in ```UriUploader```